vllm/engine/llm_engine.py

```diff
@@ -885,3 +904,43 @@ def list_loras(self) -> List[int]:
 
     def check_health(self) -> None:
         self.model_executor.check_health()
+
+    def create_trace_span(self, seq_group: SequenceGroup, now: float) -> None:
+        arrival_time_nano_seconds = int(seq_group.metrics.arrival_time * 1e9)
+        with tracer.start_as_current_span(
+                "llm_request",
+                kind=SpanKind.SERVER,
+                context=seq_group.trace_context,
+                start_time=arrival_time_nano_seconds) as seq_span:
+            metrics = seq_group.metrics
+            ttft = metrics.first_token_time - metrics.arrival_time
+            e2e_time = now - seq_group.metrics.arrival_time
+            # attribute names are based on
+            # https://github.com/open-telemetry/semantic-conventions/blob/main/docs/gen-ai/llm-spans.md
```

Comment:
Thanks for the adoption of otel llm semantic convetion, this is great!

Reply:
Great. also do we have a holistic view of different observability metrics in vLLM?

I see another effort here: https://github.com/vllm-project/vllm/issues/5041

Reply:
We have this issue: https://github.com/vllm-project/vllm/issues/3616#issuecomment-2030858781